### PR TITLE
WebPageTest Pack

### PR DESCRIPTION
**Pull request by Linuturk**
Interfaces with the public WebPageTest.org instance, or your own private instance.

Requires:
 * `requests==2.6.0`
 * API Key from [WebPageTest.org](http://www.webpagetest.org/getkey.php) (Only for the public instance.)

## Configuration

 * `wpt_url` - Base URL for the WebPageTest instance to use. Defaults to `http://webpagetest.org`.
 * `key` - The API key you received from the [WebPageTest Public Instance](http://www.webpagetest.org/getkey.php). Private instances of WebPageTest do not require a key.

## Actions

 * `list_locations` - Lists the available testing locations for your instance. Takes no arguments.
 * `request_test` - Requests a test to be run. You must specify the `domain` to be tested, as well as the `location` from which to test. The location must be formatted exactly how the `list_locations` action returns the list.
 * `random_test` - Requests a test to run from a randomly selected location. You must specify the `domain` to be tested.

### CODE REVIEW

**Comment by dzimine** [comment]
LGTM.
Thanks for the contribution Justin!

File: packs/webpagetest/actions/lib/webpagetest.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            params['k'] = self.key
+        request = requests.get("{0}/getLocations.php".format(self.wpt_url),
+                               params=params)
+        locations = request.json()['data']
+        return sorted(locations.keys())
+
+    def request_test(self, domain, location):
+        """
+        Execute a test for the given domain at a specific location.
+        Optional key is required for Google's public instance.
```
Comment (Kami) [inline packs/webpagetest/actions/lib/webpagetest.py]: Slightly confused - what's the optional key, how is it provided to this method?

Comment (Kami) [inline packs/webpagetest/actions/lib/webpagetest.py]: Nvm, I just noticed it's supplied in the config.

**Comment by Kami** [comment]
But yeah, what @dzimine has said. Looking good - thanks!

**Comment by Linuturk** [comment]
Thanks!

**Pull request merged**